R/get.R

```diff
@@ -0,0 +1,228 @@
+  t <- ifelse(lengths(t) < 10, stringr::str_split(as.character(t),
+                                                  "((?=ARTICLE)|(?=ANNEX))"), t)
+  # Get articles if declared
+  if (isTRUE(article == "preamble")) {
+    p <- lapply(t, function(x) grep("^preface|^preamble", x,
+                                    ignore.case = TRUE, value = TRUE))
+    t <- ifelse(lengths(p) == 0, purrr::map_chr(t, 1), p)
+  } else if (isTRUE(article == "accession")) {
+    t <- lapply(t, function(x) {
+      grep("([^\\s]+\\s+){0,20}open for accession([^\\s]+\\s+){0,20}|
+          |([^\\s]+\\s+){0,20}accession shall be([^\\s]+\\s+){0,20}|
+          |([^\\s]+\\s+){0,20}accede to([^\\s]+\\s+){0,20}|
+          |([^\\s]+\\s+){0,20}may join([^\\s]+\\s+){0,20}|
+          |([^\\s]+\\s+){0,20}open for joining([^\\s]+\\s+){0,20}|
+          |([^\\s]+\\s+){0,20}open for signature([^\\s]+\\s+){0,20}|
+          |([^\\s]+\\s+){0,20}may become a member([^\\s]+\\s+){0,20}}|
+          |([^\\s]+\\s+){0,20}accede thereto([^\\s]+\\s+){0,20}|
+          |([^\\s]+\\s+){0,20}become parties([^\\s]+\\s+){0,20}|
+          |([^\\s]+\\s+){0,20}become a party([^\\s]+\\s+){0,20}|
+          |([^\\s]+\\s+){0,20}request accession([^\\s]+\\s+){0,20}|
+          |([^\\s]+\\s+){0,20}may be admitted([^\\s]+\\s+){0,20}",
+           x, ignore.case = TRUE, perl = TRUE, value = TRUE)
+    })
+  } else if (isTRUE(article == "termination")) {
+    t <- lapply(t, function(x) grep("shall terminate|shall remain in force|will expire on|
+                                    |concluded for a period|shall apply for|
+                                    |periode de|shall be terminated|expiration of the period|
+                                    |denunciation|terminated|shall supersede|shall.*supplant|
+                                    |shall be extended through|have denounced this convention|
+                                    |shall be dissolved|may decide.*to dissolve|may be dissolved|
+                                    |renounce its membership|may withdraw|extraordinary events|
+                                    |failure of obligation|nonperformance of obligations|
+                                    |conflict with.*jus cogens|state party existence.*come to.*end|
+                                    |incompatibility between.*agreement and UN charter|
+                                    |incompatibility between.*agreement and United Nations charter|
+                                    |in the case of war.*end|party injurious.*end.*obligations|
+                                    |may.*denounce|any member.*may.*withdraw|injured party.*end.*obligations|
+                                    |party.*may withraw|renunciation.*by.*party",
+                                    x, ignore.case = TRUE, value = TRUE))
+  } else if (isTRUE(article == "annex")) {
+    t <- lapply(t, function(x) grep("^annex", x, ignore.case = TRUE,
+                                    value = TRUE))
+  }
```

Comment:
Maybe consider making these separate functions, and then the core function calls one of them based on a switch?